src/ghci/warning_formatter.rs

```diff
@@ -0,0 +1,236 @@
+//! Warning message formatting with GHC-style coloring.
+//!
+//! This module provides functionality to colorize GHC diagnostic messages
+//! to match the output format and colors used by GHC itself.
+
+use owo_colors::OwoColorize;
+use owo_colors::Stream::Stdout;
+
+use crate::ghci::parse::Severity;
+
+/// Formats warning and error messages with GHC-compatible colors.
+pub struct WarningFormatter;
+
+// TODO: Connect it to the ANSI colors
+// that we need to preserve in https://github.com/MercuryTechnologies/ghciwatch/blob/TrackWarnings/src/ghci/parse/ghc_message/mod.rs#L153
+impl WarningFormatter {
+    /// Create a new warning formatter.
+    pub fn new() -> Self {
+        Self
+    }
+
+    /// Apply GHC-style coloring to a complete diagnostic message.
+    ///
+    /// This processes multi-line messages and applies appropriate coloring
+    /// to each line based on GHC's output patterns.
+    pub fn colorize_message(&self, message: &str, severity: Severity) -> String {
+        let lines: Vec<&str> = message.lines().collect();
+        let mut colored_lines = Vec::new();
+
+        for line in lines {
+            let colored_line = self.colorize_line(line, severity);
+            colored_lines.push(colored_line);
+        }
+
+        colored_lines.join("\n")
```

Comment:
I think you can replace most of this with a single iterator pipeline:

```
message.lines()
    .map(|line| self.colorize_line(line, severity))
    // .collect::<Vec<_>>() // Might be necessary?
    .join("\n")
```

If that doesn't work, I think you can at least avoid the first collection:

```
for line in message.lines() {
    // ... unchanged ...
}
```

Reply:
Yeah that's much cleaner. I'll push a fix for it.

Reply:
Done!